### ISSUE TITLE
Sometimes `toc.follow` doesn't move last entries into view

### ISSUE THREAD

**Issue by reregaga** [squidfunk/mkdocs-material#4960]
### Context

```yml
site_name: My Docs

nav:
  - 'Title': index.md
  - 'Lorem': second.md

theme:
  name: material
  features:
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.footer
    - toc.follow
```

`index.md`:
```
---
hide:
  navigation
---

# H1

## H2

### H3

Text.

**Text**

One | Two
---|---
One | Lorem ipsum

### H3

Text.

**Text**

One | Two
---|---
One | Lorem ipsum

... Add more headers with content

... Headings that will sometimes be hidden in the TOC
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv

### H3 - 222

Text.

**Text**

One | Two
---|---
One | Lorem ipsum

### H3 - 111

Text.

**Text**

One | Two
---|---
One | Lorem ipsum

### H3 - 000

Text.

**Text**

One | Two
---|---
One | Lorem ipsum
```

### Bug description

I use the touchpad to scroll down the page, and sometimes the table of contents does not scroll to the end if the last few chapters fit in the viewing area.

### Related links

- https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#anchor-following

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/10544180/example.zip)

### Steps to reproduce

Scroll down.

https://user-images.githubusercontent.com/76629445/215703704-f5ecf873-6604-4f68-9784-d5c5f40240e7.mp4

https://user-images.githubusercontent.com/76629445/215703754-2419f393-3072-4eea-947f-c979c3396240.mp4

### Browser

Chrome, Firefox

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and providing the excellent reproduction! It made tracking down and fixing the issue a breeze. Fixed in c5459af35. Next time I work on anchor following I'll do a little refactoring, but for the time being this should work now 😊

https://user-images.githubusercontent.com/932156/216291394-99f0e29e-f0ba-47e7-8eec-407b73a7f249.mp4

**Comment by squidfunk** [comment]
Released as part of [9.0.10](https://github.com/squidfunk/mkdocs-material/releases/tag/9.0.10).
